go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -186,21 +203,49 @@ func (h *secondaryLocalnetNetworkControllerEventHandler) IsObjectInTerminalState
 // for a secondary localnet network
 type SecondaryLocalnetNetworkController struct {
 	BaseSecondaryLayer2NetworkController
+	openvSwitch     *vswitchd.OpenvSwitch
+	ovsClientCancel context.CancelFunc
+	nodeName        string
 }
 
 // NewSecondaryLocalnetNetworkController create a new OVN controller for the given secondary localnet NAD
 func NewSecondaryLocalnetNetworkController(
 	cnci *CommonNetworkControllerInfo,
 	netInfo util.NetInfo,
 	networkManager networkmanager.Interface,
-) *SecondaryLocalnetNetworkController {
+) (*SecondaryLocalnetNetworkController, error) {
+	return newSecondaryLocalnetNetworkController(cnci, netInfo, networkManager, nil)
+}
+
+func newSecondaryLocalnetNetworkController(
+	cnci *CommonNetworkControllerInfo,
+	netInfo util.NetInfo,
+	networkManager networkmanager.Interface,
+	ovsClient client.Client,
+) (*SecondaryLocalnetNetworkController, error) {
 
 	stopChan := make(chan struct{})
 
+	ctx, cancel := context.WithCancel(context.Background())
+	if ovsClient == nil {
```

Comment:
There is something weird here if `ovsClient != nil` we are going to build a cancel context not related to ovsClient, I understand that this is for unit test and all, but feels like cancel context from ovsclient has to be passed with it or build if nil.

Reply:
we dont cancel it for UT (UT cancel it via a chan, shared between the fake server and client)
it is because otherwise the linter fails for context that wasnt cancel.
We do it only if the current function have failures, otherwise we set it in the struct, and Stop function will call it.

Reply:
we initialize it once, instead calling it everytime,
there is quite lot of logic there
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/dd2a70c928d362a8d471352e7ec6e07164064e9f/go-controller/pkg/libovsdb/libovsdb.go#L231
i think it is nicer as now (Miguel suggested that)

Reply:
Ahh wait a minute... looks like we don't use the client anymore after initializing it, we just retrieve the OpenSwitch stuff.

This means that we can be not detecting the problem since we are not doing GetOpenvSwitch at the check function.

So even if they create the ovn-mapping after controller starts it's not going to be detected.

Reply:
i made sure, we do get updates in real time with the current PR
the current logic is good enough in this aspect in production code

Reply:
That's weird, like OpenvSwitch object get updated behind the scenes ? if this is the case then I am fine, although if feels a little dangerous.

Reply:
yep it does,
i tried to run the current solution, let it fail once, add the mapping 
recreate the pod, and it now found the ovs mapping

Reply:
about ovsClient versus openvSwitch
GetOpenvSwitch does a list, which returns a pointer to the cache, it can answer your question about how it updates,
Miguel suggested and i agree that we can just skip that
please decide, my vote is keep as is, we don't need to be over afraid imho

Reply:
I understand how it updates it but that means we depend on the implementation, like for example they can change the implementation in the future and return new pointers instead of exposing internal cache.

Reply:
let's do whatever it is other people are doing. That seems to be what @qinqon 's suggesting, i.e. calling `GetOpenvSwitch` .